project/BuildSettings.scala

```diff
@@ -0,0 +1,87 @@
+// SBT
+import sbt._
+import Keys._
+
+// sbt-assembly
+import sbtassembly._
+import sbtassembly.AssemblyKeys._
+
+/**
+  * Common settings-patterns for Snowplow apps and libraries.
+  * To enable any of these you need to explicitly add Settings value to build.sbt
+  */
+object BuildSettings {
+
+  lazy val buildSettings = Seq[Setting[_]](
+    scalacOptions := Seq(
```

Comment:
These are compiler options. You can for example tweak compiler to support particular features that are considered "dangerous" by creators or you can fail whole compilation because you accidentally used deprecated feature.

Helpful, but really important for us now.